api/hypershift/v1beta1/controlplanecomponent_types.go

```diff
@@ -0,0 +1,83 @@
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func init() {
+	SchemeBuilder.Register(func(scheme *runtime.Scheme) error {
+		scheme.AddKnownTypes(SchemeGroupVersion,
+			&ControlPlaneComponent{},
+			&ControlPlaneComponentList{},
+		)
+		return nil
+	})
+}
+
+const (
+	// ControlPlaneComponentAvailable indicates whether the ControlPlaneComponent is available.
+	ControlPlaneComponentAvailable ConditionType = "Available"
+
+	// WaitingForDependenciesReason indicates that there are unavailable dependencies blocking the ControlPlaneComponent reconcilation.
+	WaitingForDependenciesReason string = "WaitingForDependencies"
+	// ReconciliationErrorReason indicates that there was an error during the reconcilation of the ControlPlaneComponent.
+	ReconciliationErrorReason string = "ReconciliationError"
+)
+
+// ControlPlaneComponentSpec defines the desired state of ControlPlaneComponent
+type ControlPlaneComponentSpec struct {
+}
+
+type ComponentResource struct {
+	Kind string `json:"kind"`
+
+	Group string `json:"group"`
+
+	Name string `json:"name"`
+}
+
+// ControlPlaneComponentStatus defines the observed state of ControlPlaneComponent
+type ControlPlaneComponentStatus struct {
+	// Resources is a list of the resources reconciled by this component.
+	// +optional
+	Resources []ComponentResource `json:"resources,omitempty"`
+
+	// Version reports the current version of this component.
+	Version string `json:"version,omitempty"`
```

Comment:
I'm wondering if it makes sense to report multiple versions here. For example, for kube-apiserver clusteroperator on a standalone cluster, I get:

```
  versions:
  - name: raw-internal
    version: 4.18.0-0.nightly-2024-10-01-062804
  - name: kube-apiserver
    version: 1.30.3
  - name: operator
    version: 4.18.0-0.nightly-2024-10-01-062804
```

Reply:
I think we are more interested in the version of the components we reconcile/deploy than the operands they deploy in our case, right? imo I think having multiple versions reported might be confusing.

Reply:
If this is optional, we should have +optional in the godoc

Reply:
If not optional, then it should not have omitempty